gst/nnstreamer/tensor_common.c

```diff
@@ -1296,6 +1296,82 @@ gst_tensor_time_sync_get_current_time (GstCollectPads * collect,
   return FALSE;
 }
 
+/**
+ * @brief gst_tensor_time_sync_buffer_from_collectpad's internal function.
+ * SYNC helper "SLOWEST"
+ */
+static gboolean
+gst_tensor_time_sync_buffer_from_collectpad_SYNC_SLOWEST (GstBuffer ** buf,
+    GstClockTime current_time, GstTensorCollectPadData * pad,
+    GstCollectPads * collect, GstCollectData * data, gboolean * need_buffer)
+{
+  if (*buf != NULL) {
+    if (GST_BUFFER_PTS (*buf) < current_time) {
+      gst_buffer_unref (*buf);
+      if (pad->buffer != NULL)
+        gst_buffer_unref (pad->buffer);
+      pad->buffer = gst_collect_pads_pop (collect, data);
+      *need_buffer = TRUE;
+      return FALSE;
+    }
+
+    if (pad->buffer != NULL &&
+        (ABS (GST_CLOCK_DIFF (current_time,
+                    GST_BUFFER_PTS (pad->buffer))) <
+            ABS (GST_CLOCK_DIFF (current_time, GST_BUFFER_PTS (*buf))))) {
+      gst_buffer_unref (*buf);
+      *buf = pad->buffer;
+    } else {
+      gst_buffer_unref (*buf);
+      *buf = gst_collect_pads_pop (collect, data);
+      if (pad->buffer != NULL)
+        gst_buffer_unref (pad->buffer);
+      pad->buffer = *buf;
+    }
+  } else {
+    *buf = pad->buffer;
+  }
+  return TRUE;
+}
+
+/**
+ * @brief gst_tensor_time_sync_buffer_from_collectpad's internal function.
+ * SYNC helper "BASEPAD"
+ */
+static gboolean
+gst_tensor_time_sync_buffer_from_collectpad_SYNC_BASEPAD (GstBuffer ** buf,
```

Comment:
SLOWEST and BASEPAD functions look almost same.
Is it to reduce CC that does not integrate SLOWEST and BASEPAD?

Reply:
Updated.

Please review the new code carefully if there is no logic errors.

Reply:
There seems to be no problem with the logic!